plugins/nf-k8s/src/main/nextflow/k8s/K8sConfig.groovy

```diff
@@ -89,43 +248,30 @@ class K8sConfig implements Map<String,Object> {
     }
 
     Map<String,String> getLabels() {
-        podOptions.getLabels()
+        pod.getLabels()
     }
 
     Map<String,String> getAnnotations() {
-        podOptions.getAnnotations()
-    }
-
-    K8sDebug getDebug() {
-        new K8sDebug( (Map<String,Object>)get('debug') )
+        pod.getAnnotations()
     }
 
     boolean getCleanup(boolean defValue=true) {
-        target.cleanup == null ? defValue : Boolean.valueOf( target.cleanup as String )
+        defValue
     }
```

Comment:
There is a hidden option `k8s.cleanup` which is enabled by default, but can be disabled to not delete any pods. Normally only successful pods are deleted. Not sure if we still need this setting.